python/lsst/ts/mtmount/mtmount_csc.py

```diff
@@ -1659,25 +1651,151 @@ async def do_stopTracking(self, data):
+            try:
+                park_position = ParkPosition(data.position)
+            except ValueError:
+                valid_park_positions = ", ".join(
+                    [f"{position!r}" for position in ParkPosition]
+                )
+                raise ValueError(
+                    f"Invalid park position {data.position}. Must be one of {valid_park_positions}."
+                )
+
+            park_position_altaz = self.config.park_positions[park_position.name.lower()]
+
+            # load park settings
+            await self.handle_apply_settings_set(
+                restore_defaults=True,
+                settings_to_apply=self.config.park_settings,
+            )
+
+            # move telescope to the park position
+            self.log.info(f"Moving telescope to {park_position.name} park position.")
 
-        raise salobj.ExpectedError("Command not implemented yet.")
+            cmd_futures = await self.send_command(
+                commands.BothAxesMove(
+                    azimuth=park_position_altaz["azimuth"],
+                    elevation=park_position_altaz["elevation"],
+                ),
+                do_lock=True,
+            )
+            await cmd_futures.done
+            # leaving the telescope with the park settings loaded
+            # because this is needed to unpark it.
 
-    async def _do_restoreDefaultSettings(self, data):
+    async def do_restoreDefaultSettings(self, data):
         self.assert_enabled()
 
-        raise salobj.ExpectedError("Command not implemented yet.")
+        async with self.in_progress_loop(
+            ack_in_progress=self.cmd_restoreDefaultSettings.ack_in_progress, data=data
+        ):
+            await self.handle_apply_settings_set(
+                restore_defaults=True,
+                settings_to_apply=[],
+            )
 
-    async def _do_unpark(self, data):
+    async def do_unpark(self, data):
         self.assert_enabled()
 
-        raise salobj.ExpectedError("Command not implemented yet.")
+        async with self.in_progress_loop(
+            ack_in_progress=self.cmd_park.ack_in_progress, data=data
+        ):
+            # move telescope to the unparked position
+
+            unpark_elevation = None
+            unpark_azimuth = None
+
+            async with salobj.Remote(
+                self.domain,
+                "MTMount",
+                include=["elevation", "azimuth"],
+                readonly=True,
+            ) as mtmount_remote:
+                current_elevation = await mtmount_remote.tel_elevation.next(
+                    flush=True, timeout=TELEMETRY_START_TIMEOUT
+                )
+                current_azimuth = await mtmount_remote.tel_azimuth.next(
+                    flush=True, timeout=TELEMETRY_START_TIMEOUT
+                )
+                unpark_elevation = (
+                    current_elevation.actualPosition
+                )  # this is going to be redacted furthermore
```

Comment:
What does the comment mean? Will there be future improvements? If yes, please open a ticket for that and add a TODO referring to that ticket here.

Reply:
I updated the comment. It is just to say that it is setting the `unpark_elevation` here to match the current elevation, but later we will change it to be a value inside the operation limit, just bellow or above the max/min limits.